File: scripts/generate-il3.mjs
import fs from 'fs/promises';
import path from 'path';

const TOKEN = process.env.REPLICATE_API_TOKEN;
const OUT = '/Users/yikargecer/Desktop/fillys-website/public/media';

const items = [
  {
    slug: 'vanilya-il3',
    prompt: 'minimalist watercolor illustration of a vanilla orchid flower with one vanilla bean pod, isolated on white background, loose soft brush strokes, warm cream and golden palette, botanical food art, no shadow, no text',
  },
  {
    slug: 'pistachio-il3',
    prompt: 'minimalist watercolor illustration of three pistachio nuts cracked open showing vibrant green, isolated on white background, loose soft brush strokes, fresh green and beige palette, botanical food art, no shadow, no text',
  },
  {
    slug: 'matcha-il3',
    prompt: 'minimalist watercolor illustration of a small heap of matcha powder with two green tea leaves beside it, isolated on white background, loose soft brush strokes, deep moss green palette, botanical food art, no shadow, no text',
  },
  {
    slug: 'karamel-il3',
    prompt: 'minimalist watercolor illustration of caramel sauce drizzle drops with three whole coffee beans, isolated on white background, loose soft brush strokes, warm amber and dark brown palette, botanical food art, no shadow, no text',
  },
  {
    slug: 'beyaz-cikolata-il3',
    prompt: 'minimalist watercolor illustration of two broken white chocolate squares with delicate cocoa flower, isolated on white background, loose soft brush strokes, ivory and warm cream palette, botanical food art, no shadow, no text',
  },
];

const sleep = (ms) => new Promise((r) => setTimeout(r, ms));

async function post(url, body) {
  const res = await fetch(url, {
    method: 'POST',
    headers: {
      Authorization: `Bearer ${TOKEN}`,
      'Content-Type': 'application/json',
      Prefer: 'wait=90',
    },
    body: JSON.stringify(body),
  });
  return res.json();
}

async function poll(id) {
  for (let i = 0; i < 200; i++) {
    await sleep(3000);
    const res = await fetch(`https://api.replicate.com/v1/predictions/${id}`, {
      headers: { Authorization: `Bearer ${TOKEN}` },
    });
    const d = await res.json();
    if (d.status === 'succeeded') return d;
    if (d.status === 'failed') throw new Error('Failed: ' + d.error);
    process.stdout.write('.');
  }
  throw new Error('Timeout');
}

async function save(url, filepath) {
  const res = await fetch(url);
  const buf = await res.arrayBuffer();
  await fs.writeFile(filepath, Buffer.from(buf));
}

async function main() {
  for (const item of items) {
    console.log(`\n▶ ${item.slug}`);

    // 1. Generate with Flux-dev
    let gen = await post('https://api.replicate.com/v1/models/black-forest-labs/flux-schnell/predictions', {
      input: {
        prompt: item.prompt,
        num_outputs: 1,
        aspect_ratio: '1:1',
        output_format: 'png',
        output_quality: 95,
      },
    });
    if (gen.status !== 'succeeded') {
      console.log('  polling generation...');
      gen = await poll(gen.id);
    }
    const imgUrl = Array.isArray(gen.output) ? gen.output[0] : gen.output;
    console.log(`\n  ✓ generated`);

    // 2. Remove background with rembg
    let bg = await post('https://api.replicate.com/v1/models/cjwbw/rembg/predictions', {
      input: { image: imgUrl },
    });
    if (bg.status !== 'succeeded') {
      console.log('  polling rembg...');
      bg = await poll(bg.id);
    }
    const finalUrl = bg.output;
    console.log(`  ✓ background removed`);

    // 3. Save
    const filepath = path.join(OUT, `${item.slug}.png`);
    await save(finalUrl, filepath);
    console.log(`  ✓ saved → ${item.slug}.png`);
  }
  console.log('\n✅ All done!');
}

main().catch(console.error);

File: scripts/generate-il3.py
import os
import io
import time
import replicate
from rembg import remove
from PIL import Image

TOKEN = os.environ.get("REPLICATE_API_TOKEN")
OUT = "/Users/yikargecer/Desktop/fillys-website/public/media"

items = [
    ("vanilya-il3",        "minimalist watercolor illustration of a vanilla orchid flower with one vanilla bean pod, isolated on white background, loose soft brush strokes, warm cream golden palette, botanical food art, no shadow"),
    ("pistachio-il3",      "minimalist watercolor illustration of three pistachio nuts cracked open showing vibrant green interior, isolated on white background, loose soft brush strokes, fresh green beige palette, botanical food art, no shadow"),
    ("matcha-il3",         "minimalist watercolor illustration of matcha powder heap with two green tea leaves, isolated on white background, loose soft brush strokes, deep moss green palette, botanical food art, no shadow"),
    ("karamel-il3",        "minimalist watercolor illustration of caramel sauce drizzle drops with three coffee beans, isolated on white background, loose soft brush strokes, warm amber brown palette, botanical food art, no shadow"),
    ("beyaz-cikolata-il3", "minimalist watercolor illustration of two broken white chocolate squares with a small cocoa flower, isolated on white background, loose soft brush strokes, ivory warm cream palette, botanical food art, no shadow"),
]

client = replicate.Client(api_token=TOKEN)

def run_with_retry(model, input_data, retries=5, wait=15):
    for attempt in range(retries):
        try:
            return client.run(model, input=input_data)
        except Exception as e:
            if '429' in str(e) and attempt < retries - 1:
                print(f"  rate limit — {wait}s bekleniyor...", flush=True)
                time.sleep(wait)
            else:
                raise

for slug, prompt in items:
    print(f"\n▶ {slug}", flush=True)
    try:
        # Step 1: Flux-schnell ile üret
        print("  generating...", flush=True)
        gen_output = run_with_retry(
            "black-forest-labs/flux-schnell",
            {
                "prompt": prompt,
                "num_outputs": 1,
                "aspect_ratio": "1:1",
                "output_format": "png",
                "output_quality": 95,
            }
        )
        # FileOutput'tan bytes oku
        file_output = gen_output[0] if isinstance(gen_output, list) else gen_output
        img_bytes = file_output.read()
        print("  ✓ generated", flush=True)

        # Step 2: Lokal rembg ile arka planı kes
        print("  removing background (local rembg)...", flush=True)
        input_img = Image.open(io.BytesIO(img_bytes)).convert("RGBA")
        output_img = remove(input_img)
        print("  ✓ background removed", flush=True)

        # Step 3: Kaydet
        filepath = f"{OUT}/{slug}.png"
        output_img.save(filepath, "PNG")
        print(f"  ✓ saved → {slug}.png", flush=True)

        # Rate limit koruma
        time.sleep(12)

    except Exception as e:
        print(f"  ✗ error: {e}", flush=True)

print("\n✅ Done!")
